### DISCUSSION TITLE
Integrity comparison use case

### DISCUSSION THREAD

**Discussion by patrickwolf** [pkolaczk/fclones#199] [Ideas]
Would be great if fclones could 
- make a snapshot of a folder structure and save it to a txt file
- and then later compare the folder structure to the snapshot text file and list what changed

This would allow for use cases like 
- auditing and integrity check
- comparing folder structures after file transfer or backup restore 

I know its not the primary use case and that it would require creating a checksum of each file but for many who use fclones to manage a larger folder structure they already have the cache file which might contain a large number of the files already (ie i have 100TB in the cache file right now)

The text file structure would just be
relative file name, hash algo, hash, date/time and potentially inode

**Comment by kapitainsky** [comment]
There is already tool for this:

https://github.com/jessek/hashdeep

and use Audit mode - https://md5deep.sourceforge.net/start-hashdeep.html

It does exactly what you described
